### PR TITLE
Feature: manage products rework

### PR DESCRIPTION
**Pull request by JustSamuel**
Manage Products View and Components Rework

# Description
This PR completely overhauls the product management by merging `ContainerAssignProductDialog`, `ProductCreateComponent` and `ProductDialogComponent` into a single, unified component. 

To this end the PR also contains a rework of the container and product store which bring about easy deep reactivity and easier lazy loading. 

## Details

The `ContainerActionsDialog` handles the creation and editing of the containers:
The `AddContainerDialgoComponent.vue` was merged into the `ContainerActionsDialog`.
image
image
It also contains a tooltip, similar to the product creation dialog, which highlights the detail that container owners receive money, not product owners:
image

It is possible to edit containers by pressing the "edit pencil":
image

The `ContainerAssignProductDialog`, `ProductCreateComponent` and `ProductDialogComponent` are merged in a single, unified `ProductContainerOperationsDialog`.
Here is how it looks when selecting a product in a container:
image
Example of mobile view:
Screen Shot 2024-06-19 at 15 36 01

You can also directly add a product to a container using the + in the containers' product list:
image
The dropdown on the top left allows a user to add already defined products:
image

When a user is typing a product name similar to one already existing, they will be prompted with the option to select it:
image
This tip will also show products that are in the container already, to make sure that the user is aware of the products existence. Clicking will then not allow the user to add the product, but does inform them:
image

You can also create products without linking it to a container at the product overview:
image
The inline row editing has been removed in favor of re-using the dialog.
image
(Note the same tooltip as in the create container dialog)

The product and container store have also been reworked to allow for nice deep reactivity. In almost all cases this can be achieved using: 
```const containers = computed<ContainerInStore[]>(() => Object.values(containerStore.getAllContainers));```
When using a single container or product, a simple ref would suffice.

## Related issues/external references
#119 
## Types of changes
- Bug fix _(non-breaking change which fixes an issue)_
- New feature _(non-breaking change which adds functionality)_
- Documentation improvement

### CODE REVIEW

**Comment by SuperVK** [comment]
When adding an existing product to a container, the modal doesn't close

Changing the picture does require a reload of the page to be shown in the general overview.

Would be nice if your cursor would show pointer on hover of products in containers.

Sometimes I had issues with created products, where the modal wouldn't close, but turns out it did make the product. Can't reproduce anymore atm tho

image
Would it be possible to remove this form validation when selecting a product?

image
Might be nice to give price, alc, and VAT some more room in favor of name.

Changing properties of a product does not work from the container overview.

Do we maybe want to change the edit mode in the product overview to also give the same modal as editing in the container overview? Preventing issues such as below:
image


Updating most properties from editing in product overview do reflect in the container overview, but VAT does not.

Might be nice to have the ability to make containers from the container overview

**Comment by SuperVK** [comment]
Oh btw besides all these comments: Already very good looking, the reactive updating is *chefs kiss*

File: apps/dashboard/src/components/ContainerCardComponent.vue [hunk 1]
```diff
@@ -1,38 +1,43 @@
...
-      </TabPanel>
-      <TabPanel>
-        <template #header>
-            <span class="p-tabview-title createContainerButton">{{ $t('containersOverview.newContainer') }}</span>
-        </template>
-      </TabPanel>
-    </TabView>
-    <Accordion :activeIndex="0" class="block md:hidden w-full">
+<!--    TODO re-add newContainer button -->
+<!--    <TabView :active-index=1 class="hidden md:block" :scrollable="true">-->
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/components/ContainerCardComponent.vue]: Commented Code

File: apps/dashboard/src/components/ContainerCardComponent.vue [hunk 2]
```diff
@@ -1,38 +1,43 @@
...
-        <ProductGridComponent :container="container" :products="container.products"/>
-      </TabPanel>
-      <TabPanel>
-        <template #header>
-            <span class="p-tabview-title createContainerButton">{{ $t('containersOverview.newContainer') }}</span>
-        </template>
-      </TabPanel>
-    </TabView>
-    <Accordion :activeIndex="0" class="block md:hidden w-full">
+<!--    TODO re-add newContainer button -->
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/components/ContainerCardComponent.vue]: Issue or fix

**Review by CodeNamedRobin — CHANGES_REQUESTED** [review]
Overall this rework is pretty great, however still unusable on mobile.

**Comment by CodeNamedRobin** [comment]
Having products appear in one column might not be the greatest use of space.
image
Also, in the products table things overlap, and the create button is outside of the table
image
The ProductCreateModal is outside of the screen currently.
image

**Comment by CodeNamedRobin** [comment]
You can't choose containers on create pos view
image

**Comment by SuperVK** [comment]
The create products button on all products has broken when it is the first thing you click after reloading the page, after which the whole page goes into a panick, and nothing works anymore. If you try to edit a product first, and then click create, it will show the last edited product.

VAT is still showed as the default from the product category related to the product, even if the database information for that product is different from it.

image
The create button could use some space from the containers

**Comment by SuperVK** [comment]
Not that high priority but still nice: Right now the behavior when the network is offline is kinda wonky, the modal does not close, which is somewhat fair, but it might be nice to handle this with the toast services. Furthermore, the modal does refuse to close, but the product _is_ added to the container, until refreshed atleast. 

The container edit/create modal is missing some translations

**Comment by JustSamuel** [comment]
>  The create button could use some space from the containers
I like it as a kinda "bookmark"

**Comment by SuperVK** [comment]
> > The create button could use some space from the containers
> > I like it as a kinda "bookmark"

I see what you mean, but that is inconsistent with the styling in the rest of the app(, and I think for the bookmark to look good you want to remove the radius on the bottom).

File: apps/dashboard/src/components/ProductCardComponent.vue [hunk 3]
```diff
@@ -0,0 +1,220 @@
+<template>
+  <CardComponent header="all products" class="w-full">
```
Comment (SuperVK) [inline apps/dashboard/src/components/ProductCardComponent.vue]: Needs translations

**Review by SuperVK — APPROVED** [review]
Missing one translation, but besides that, I think it's pico bello!

**Review by SuperVK — APPROVED** [review]
LGTM!

**Pull request merged**